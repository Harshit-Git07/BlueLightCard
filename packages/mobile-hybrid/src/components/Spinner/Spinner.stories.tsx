import { Meta, StoryFn } from '@storybook/react';
import Spinner from './Spinner';
import tokenMigrationDecorator from '@storybook/tokenMigrationDecorator';

const componentMeta: Meta<typeof Spinner> = {
  title: 'Spinner',
  component: Spinner,
  decorators: [tokenMigrationDecorator],
};

const DefaultTemplate: StoryFn<typeof Spinner> = (args) => <Spinner {...args} />;

export const Default = DefaultTemplate.bind({});

Default.args = {};

export default componentMeta;
