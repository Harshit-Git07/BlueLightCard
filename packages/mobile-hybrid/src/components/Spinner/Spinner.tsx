const Spinner = () => {
  return (
    <div className="p-3 rounded bg-white dark:bg-gray-700 shadow-lg">
      <svg
        className="spinner"
        width="60px"
        height="60px"
        viewBox="0 0 66 66"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          className="path"
          fill="none"
          strokeWidth="6"
          strokeLinecap="round"
          cx="33"
          cy="33"
          r="30"
        ></circle>
      </svg>
    </div>
  );
};

export default Spinner;
