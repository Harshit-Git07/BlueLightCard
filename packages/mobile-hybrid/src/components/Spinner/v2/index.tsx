import { FC } from 'react';
import { cssUtil } from '@/utils/cssUtil';

const Spinner: FC = () => {
  const spinnerClass = cssUtil([
    'path',
    'stroke-spinner-colour-light dark:stroke-spinner-colour-dark',
  ]);
  return (
    <div role="progressbar" className="p-3 rounded shadow-lg">
      <svg
        className="spinner"
        width="60px"
        height="60px"
        viewBox="0 0 66 66"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          className={spinnerClass}
          fill="none"
          strokeWidth="6"
          strokeLinecap="round"
          cx="33"
          cy="33"
          r="30"
        ></circle>
      </svg>
    </div>
  );
};

export default Spinner;
