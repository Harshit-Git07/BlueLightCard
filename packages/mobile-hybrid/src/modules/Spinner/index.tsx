import { FC } from 'react';
import SpinnerContainer from './components/SpinnerContainer';

const Spinner: FC = () => {
  return <SpinnerContainer />;
};

export default Spinner;
