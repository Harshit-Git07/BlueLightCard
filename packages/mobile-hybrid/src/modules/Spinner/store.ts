import { atom } from 'jotai';

export const spinner = atom(true);
