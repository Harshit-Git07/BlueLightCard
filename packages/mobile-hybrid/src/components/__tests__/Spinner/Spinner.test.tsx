import { render } from '@testing-library/react';
import Spinner from '@/components/Spinner/Spinner';

describe('Spinner Component', () => {
  it('should render correctly', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
