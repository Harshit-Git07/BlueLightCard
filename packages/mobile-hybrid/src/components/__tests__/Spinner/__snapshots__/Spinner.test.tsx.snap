// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render correctly 1`] = `
<div>
  <div
    class="p-3 rounded shadow-lg"
    role="progressbar"
  >
    <svg
      class="spinner"
      height="60px"
      viewBox="0 0 66 66"
      width="60px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="path stroke-spinner-colour-light dark:stroke-spinner-colour-dark"
        cx="33"
        cy="33"
        fill="none"
        r="30"
        stroke-linecap="round"
        stroke-width="6"
      />
    </svg>
  </div>
</div>
`;
